runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/control/FnApiControlClientPoolService.java

```diff
@@ -105,12 +105,38 @@ public static FnApiControlClientPoolService offeringClientsToPool(
     return newClient.asResponseObserver();
   }
 
+  @Override
+  public void getProcessBundleDescriptor(
```

Comment:
Please add failure tests to FnApiControlClientPoolService for NOT_FOUND and missing worker id.